sway-core/src/ir_generation/function.rs

```diff
@@ -613,8 +613,10 @@ impl<'a> FnCompiler<'a> {
             ty::TyExpressionVariant::ConfigurableExpression {
                 decl: const_decl, ..
             } => self.compile_config_expr(context, const_decl, span_md_idx),
-            ty::TyExpressionVariant::ConstGenericExpression { decl, .. } => {
-                let value = decl.value.as_ref().unwrap();
+            ty::TyExpressionVariant::ConstGenericExpression { decl, span, .. } => {
+                let Some(value) = decl.value.as_ref() else {
+                    panic!("Const generic not materialized: {:?}", span);
```

Comment:
This is a discussion we had internally a few times, but never wrote down the guidelines, when to `panic!` and when to emit `CompilerError::Internal`. For the cases when we expect the previous part of the compile pipeline do something and that doesn't happen, the proposal was to go with emitting `CompileError::Internal`. This is the case in the rest of this file, so I am for having `CompileError::Internal` here as well.